packages/plugin-compiler/src/main/kotlin/io/realm/compiler/AccessorModifierIrGeneration.kt

```diff
@@ -154,7 +155,15 @@ class AccessorModifierIrGeneration(private val pluginContext: IrPluginContext) {
                 val propertyTypeRaw = declaration.backingField!!.type
                 val propertyType = propertyTypeRaw.makeNotNull()
                 val nullable = propertyTypeRaw.isNullable()
+                val excludeProperty = declaration.backingField!!.hasAnnotation(FqNames.IGNORE_ANNOTATION) ||
+                    declaration.backingField!!.hasAnnotation(FqNames.TRANSIENT_ANNOTATION) ||
+                    declaration.backingField!!.isFinal
```

Comment:
What is the motivation for this `isFinal`?

Reply:
It identifies immutable properties `val`. I thought of excluding constants from persistence because their value does not change and it is already persisted in the model definition.

But now you raise the point, with sync we could have a use case where we have some field mutable on an app, and immutable on others.

Reply:
I would have reservations about doing this as well. We already have the concept of read-only fields, like primary keys, which we do need to store in the DB.